Add documentation for ResizePolicy field

Sapthagiri777 opened the pull request:
<!--  
Thanks for sending a pull request! Please take a look at the contribution guidelines:  
https://git.k8s.io/community/contributors/guide/first-contribution.md  
-->

#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

This PR corrects the API documentation for the `ResizePolicy` field in `core/v1/types.go`.

Previously, the documentation may have implied that `resizePolicy` could apply to ephemeral containers. This update clarifies that the field only applies to regular and init containers, not ephemeral ones. This clarification is important to help developers understand correct usage and avoid confusion when working with container resize behaviors.

#### Which issue(s) this PR fixes:

Fixes #128384

#### Special notes for your reviewer:

This is my first contribution to the Kubernetes project, and I’m currently learning under the guidance of my mentor, **@jcano-cano (Javier Cano Cano)**, who kindly offered to review this PR.  
Please let me know if any changes or improvements are needed. Thank you for the opportunity!

/cc @jcano-cano

#### Does this PR introduce a user-facing change?

```release-note
NONE

Sapthagiri777 committed:
Add documentation for ResizePolicy field in core/v1 types

Sapthagiri777 commented:
Hi @jcano-cano, I've corrected the release note and the PR is ready for review. Please let me know if any further changes are needed. Thank you!

pacoxu commented:
The PR description are in a code block. Can you remove the starting ```?

/kind documentation

Sapthagiri777 committed:
Enhance: improved failure messages in storage tests (#10347)

Sapthagiri777 commented:
Hi @brianmcarey and @qinqon 👋,

I’ve worked on this PR as a student contributor to improve a test failure message in the storage e2e tests (#131442).

Could I kindly request your review and a `/lgtm` if the changes look good?  
Once reviewed, I’ll assign `@msau42` for final `/approve`.

Your feedback and time mean a lot — thank you   
— @Sapthagiri777

Sapthagiri777 commented:
Hi @msau42 @pacoxu,

Kind reminder — I’ve addressed the formatting issue and updated the PR description as requested.  
This is part of my Google Summer of Code (GSoC) 2025 contribution, and the contributor announcements are scheduled for May 8. I’d be grateful if you could take a moment to review it and let me know if anything else is needed.

Thank you for your time and support! 

/assign @msau42

pacoxu commented:
/cc @kannon92  
/retitle Add documentation for ResizePolicy field
/test pull-kubernetes-verify

pacoxu commented:
/release-note-none

pacoxu commented on staging/src/k8s.io/api/core/v1/types.go:
```diff
@@ -2830,6 +2830,11 @@ type Container struct {
+	// can be resized while the container is running. This is an optional field that specifies
+	// one or more policies that control if and how container resource requests can be updated
+	// dynamically.
+	// +optional
```
```suggestion
	// ResizePolicy defines how resources allocated to a container (such as CPU and memory)
	// can be resized while the container is running. This is an optional field that specifies
	// one or more policies that control if and how container resource requests can be updated
	// dynamically.
	// +featureGate=InPlacePodVerticalScaling
	// +optional
	// +listType=atomic
```

`+optional` is dup.
